public/scripts/upload.js

```diff
@@ -788,6 +803,38 @@ function main() {
     },
   );
 
+  // observe fileListViewModel: imageUploads[].generatedMockups[].length
+  mobx.reaction(
+    () =>
+      viewModel.fileList.imageUploads.map(
+        (imageUpload) => imageUpload.generatedMockups.length,
+      ),
+    async () => {
+      if (viewModel.isGeneratingMockup) {
+        const imageUploads = viewModel.fileList.imageUploads;
+        let generatedMockups = [];
+        for (let i = 0; i < imageUploads.length; i += 1) {
+          generatedMockups = [
+            ...generatedMockups,
+            ...imageUploads[i].generatedMockups,
+          ];
+          if (
+            imageUploads[i].generatedMockups.length < viewModel.orientationsNum
+          ) {
+            return;
+          }
+        }
+
+        window.localforage
+          .setItem("pictureArray", generatedMockups)
+          .then(() => {
+            window.location.href =
+              "/download/?deviceId=" + window.workerDeviceId;
+          });
+      }
+    },
```

Comment:
Can extract this as 2 functions?

1. call `onNewGeneratedMockup`
2. in `onNewGeneratedMockup`, call `navigateToDownloadPAge`